ql/time/calendars/newzealand.cpp

```diff
@@ -62,6 +62,26 @@ namespace QuantLib {
             // Boxing Day, December 26th (possibly Monday or Tuesday)
             || ((d == 26 || (d == 28 && (w == Monday || w == Tuesday)))
                 && m == December))
+            // Matariki, it happens on Friday in June or July
+            // official calendar released by the NZ government for the
+            // next 30 years
+            || (d == 20 && m == June && y == 2025)
+            || (d == 21 && m == June && (y == 2030 || y == 2052))
+            || (d == 24 && m == June && (y == 2022 || y == 2033 || y == 2044))
+            || (d == 25 && m == June && (y == 2027 || y == 2038 || y == 2049))
+            || (d == 28 && m == June && y == 2024)
+            || (d == 29 && m == June && (y == 2035 || y == 2046))
+            || (d == 30 && m == June && y == 2051)
+            || (d == 2  && m == July && y == 2032)
+            || (d == 3  && m == July && (y == 2043 || y == 2048))
+            || (d == 6  && m == July && (y == 2029 || y == 2040))
+            || (d == 7  && m == July && (y == 2034 || y == 2045))
+            || (d == 10 && m == July && (y == 2026 || y == 2037))
+            || (d == 11 && m == July && (y == 2031 || y == 2042))
+            || (d == 14 && m == July && (y == 2023 || y == 2028))
+            || (d == 15 && m == July && (y == 2039 || y == 2050))
+            || (d == 18 && m == July && y == 2036)
+            || (d == 19 && m == July && (y == 2041 || y == 2047))
```

Comment:
...and add it here at the end.

Reply:
Arrgh... Thanks Luigi and sorry for the loss of time! Ok noted: before pushing any PR to master (even those that look very simple) I have to code it and be sure that it compiles successfully in my branch locally. Only after local branch build and test is successful, then I will push it to the main branch.

Reply:
No problem 😄

Reply:
Thanks Luigi! This time after coding I built the branch and it was ok. Tested the behaviour of all the 31 dates and it looks good. I attach the test file. Have a nice evening.
[NZMatariki.cpp.txt](https://github.com/lballabio/QuantLib/files/10421316/NZMatariki.cpp.txt)